### ISSUE TITLE
Crafter eats my stuff

### ISSUE THREAD

**Issue by dominx** [McJty/RFTools#276]
I'm playing Horizons Daybreaker 1.0.2 with rftools 3.10.

Sometimes the crafter consumes resources and doesn't produce anything.
I managed to narrow it down somewhat: [screenshot](http://i.imgur.com/2EqCRsh.png)
When I put the 2 osmium and 4 alloy it needs the tablets and said items disappear and the two cubes remain unaffected. It works fine when there is no red cube besides the green one. The mekanism cubes are non-stackable.

**Comment by McJty** [comment]
That is very weird. Are you sure nothing is extracting the output from the crafter?

**Comment by dominx** [comment]
Yes, it bordered only a cable and some Mek machine. They can't pull. I also possibly ran into similar problem before also with crafting Mek solar panels but can't be sure what happened as I didn't have the ingredients carefully counted.
It looks as if the crafter somehow tried to add the new red cube to existing stack, which didn't work because the item is non stackable. There might be something off in Mek. As much as I like it, it either has more than average share of bugs or I have simply very bad luck with it.

**Comment by dominx** [comment]
I've just tested with rftools 3.30 and problem is still there.

**Comment by McJty** [comment]
Can you describe the recipes more exactly? I'm not 100% sure yet what you're doing to reproduce this.

**Comment by dominx** [comment]
I've done 3.30 testing in creative test world. 
[Tablet](http://i.imgur.com/DCFR92t.png) [Advanced Cube](http://i.imgur.com/ND2WRBR.png)
I've set recipes for Basic and Advanced Cube and the Energy Tablet. Then I've put in enough stuff to make one of each cubes and enough to upgrade the basic cube to advanced with just redstone missing. When I add 4 redstone a tablet gets made. With another four redstone another tablet is made flashes for a moment because there's enough to make upgrade the basic cube to advanced. Resources get consumed but the basic cube doesn't get upgraded, they both stay the same. If there is no advanced cube already there in one of the resource slot it works fine. If the advanced cube recipe is set to 'ext' the basic one doesn't change and the existing advanced is moved to output slot. I've also noticed that both basic and advanced empty cubes have the same id of 908:100.

BTW How does the crafter decide what to craft if it has enough resources to do two recipes sharing some ingredients?
